Source/WebCore/dom/Element.cpp

```diff
@@ -2603,6 +2614,12 @@ Node::InsertedIntoAncestorResult Element::insertedIntoAncestor(InsertionType ins
             ensureElementRareData().setEffectiveLang(lang);
     }
 
+    if (!UNLIKELY(hasRareData()) || !elementRareData()->effectiveLang()) {
+        auto langAttr = langFromAttribute();
+        if (!langAttr.isNull() && langAttr != document().documentElementLanguage())
+            setEffectiveLangInSubtree();
+    }
```

Comment:
If this element is newly connected to a document, then we must also check if it has lang attribute that's identical to document element's language, in which case we need to add it to the set of elements maintained by document to handle a case like this:
```
<html lang="zh">
<body>
<script>
const element = document.createElement('div');
element.lang = "zh";
document.body.appendChild(element);
document.documentElement = 'ja';
</script>
</html>
```